### PR TITLE
Added Auth0.Result type info

### LINKED ISSUE #317: Auth0.Result masks Swift 5 Result type
### Description

> Provide a clear and concise description of the issue, including what you expected to happen.

If you `import Auth0` and try to use the [Result](https://developer.apple.com/documentation/swift/result) type, the Auth0.Result enum wins out. Since these have different type definitions, they aren't perfectly swappable. This requires you to explicitly specify the Swift version.

For example:
```
func performRequest(completion: @escaping (Result<String,CustomNetworkError>))
```

Generates a compiler error of "Generic type 'Result' specialized with too many type parameters (got 2, but expected 1)" and "1. Generic type 'Result' declared here (Auth0.Result)".

Instead you have to:
```
func performRequest(completion: @escaping (Swift.Result<String,CustomNetworkError>))
```

Certainly easy to workaround, but I expect many devs will not realize how to call out to the default Swift version.

### Reproduction

> Detail the steps taken to reproduce this error, what was expected, and whether this issue can be reproduced consistently or if it is intermittent.
>
> Where applicable, please include:
>
> - Code sample to reproduce the issue
> - Log files (redact/remove sensitive information)
> - Application settings (redact/remove sensitive information)
> - Screenshots

Any code using `Result` and the Auth0 module:
```
import Auth0

var test: Result<String,Error>?
```

Will fail.

### Environment

> Please provide the following:

- **Version of this library used:**
"auth0/Auth0.swift" "1.17.1"
- **Version of the platform or framework used, if applicable:**
Xcode 11, Swift 5.1, iOS 12.4 as minimum target.
- **Other relevant versions (language, server software, OS, browser):**
- **Other modules/plugins/libraries that might be involved:**

### PR DESCRIPTION
**Pull request by Widcket**
### Changes

- Added information to the **Auth0.swift** documentation about how the `Auth0.Result` type may shadow Swift's.
- Switched places of the notes that were above and below the code snippet in a way that makes sense with the new paragraph.

### References
- Fixes https://github.com/auth0/Auth0.swift/issues/317

### CODE REVIEW

File: articles/libraries/auth0-swift/index.md [hunk 1]
```diff
@@ -120,15 +120,15 @@ func application(_ app: UIApplication, open url: URL, options: [UIApplicationOpe
 The first step in adding authentication to your iOS application is to provide a way for your users to log in. The fastest, most secure, and most feature-rich way to do this with Auth0 is to use <dfn data-key="universal-login">Universal Login</dfn>.
 
 ::: note
-To ensure a response that complies with <dfn data-key="openid">OpenID Connect (OIDC)</dfn>, you must either request an <dfn data-key="audience">`audience`</dfn> or enable the **OIDC Conformant** switch in your [Auth0 dashboard](${manage_url}), under **Application > Settings > Show Advanced Settings > OAuth**. For more information, refer to [How to use the new flows](/api-auth/intro#how-to-use-the-new-flows).
+If you need help between the two types of login flows, refer to [Browser-Based vs. Native Login Flows on Mobile Devices](/tutorials/browser-based-vs-native-experience-on-mobile)
```
Comment (stevehobbsdev) [inline articles/libraries/auth0-swift/index.md]: I would reword the start of this a bit. I suggest:

```suggestion
For more information on the two types of login flows, please refer to [Browser-Based vs. Native Login Flows on Mobile Devices](/tutorials/browser-based-vs-native-experience-on-mobile)
```

File: articles/libraries/auth0-swift/index.md [hunk 2]
```diff
@@ -137,8 +137,10 @@ Auth0
     }
 ```
 
+If you're using **Swift 5+** the `Auth0.Result` type may shadow Swift's `Result`. To prevent that, whenever you want to refer to Swift's type, use `Swift.Result` instead. This will be fixed in the next major version of Auth0.swift.
```
Comment (stevehobbsdev) [inline articles/libraries/auth0-swift/index.md]:
```suggestion
If you're using **Swift 5+**, `Auth0.Result` may shadow Swift's `Result` type. To prevent that, replace it with `Swift.Result` whenever you want to refer to Swift's built-in type. This will be fixed in the next major version of Auth0.swift.
```

**Review by stevehobbsdev — DISMISSED** [review]
Just a couple of non-blocking suggestions, if you choose to use them.

File: articles/libraries/auth0-swift/index.md (hunk 2 above)
Comment (Widcket) [reply articles/libraries/auth0-swift/index.md]: Thanks @stevehobbsdev!

**Review by stevehobbsdev — APPROVED** [review]
